Source/Particles/PhotonParticleContainer.cpp

```diff
@@ -99,3 +122,26 @@ PhotonParticleContainer::Evolve (int lev,
                                        t, dt);
 
 }
+
+#ifdef WARPX_QED
+// A function to initialize the Tau component according to the BW engine
+void PhotonParticleContainer::InitTauBreitWheeler()
+{
+    BL_PROFILE("PhotonParticleContainer::InitOpticalDepth");
+    //Get functor
+    auto get_opt = shr_ptr_bw_engine->build_optical_depth_functor();
+
+    //Looping over all the particles
+    int num_levels = finestLevel() + 1;
+    for (int lev=0; lev < num_levels; ++lev)
```

Comment:
Same thing as for PhysicalParticleContainer. Let me know if you want me to look into it.